components/Home/index.tsx

```diff
@@ -50,7 +56,12 @@ export const HomePage: NextPage = () => {
         .then((res) => res.json())
         .then((res) => {
           setSelectedIndex(0);
-          setResults(res.results);
+          const newResults = pipe(
+            res,
+            ResultObject.codec.decode,
+            Either.getOrElseW((e) => null)
+          );
+          setResults(newResults?.results ?? []);
```

Comment:
I think we can do that too:

```suggestion
          pipe(
            res,
            ResultObject.codec.decode,
            Either.map(({results}) => results),
            Either.getOrElseW(() => []),
            setResults
          );
```